### PR TITLE
Add vector collection metrics documentation [AI-52]

### PR DESCRIPTION
**Pull request by k-jamroz**
Documentation for metrics added in https://github.com/hazelcast/hazelcast-mono/pull/2471

### CODE REVIEW

**Review by vbekiaris — APPROVED** [review]

**Review by oliverhowell — CHANGES_REQUESTED** [review]
Thanks @vbekiaris - only minor changes (the whole page could do with a thorough review and edit tbh) but I've suggested changes.

File: docs/modules/ROOT/pages/list-of-metrics.adoc [hunk 1]
```diff
@@ -433,6 +433,196 @@ This is because the cluster has to communicate with more members, which can add
...
+|ms
+|Total latency of local set operations on the vector collection
+
+|`vectorCollection.totalSizeLatency`
+|ms
+|Total latency of local size operations on the vector collection
+
+3+a|
+The above `*latency` metrics are only measured for the members and they are not representing the overall performance of the cluster.
+We recommend monitoring the average latency for each operation, for example, `vectorCollection.totalGetLatency` / `vectorCollection.getCount` and `vectorCollection.totalSetLatency` / `vectorCollection.setCount`.
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/ROOT/pages/list-of-metrics.adoc:614]: `/` here is division to calculate average latency

File: docs/modules/ROOT/pages/list-of-metrics.adoc [hunk 2]
```diff
@@ -433,6 +433,196 @@ This is because the cluster has to communicate with more members, which can add
...
+Increased average latency is a sign that the cluster would experience performance problems, or there is a spike in the load.
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/ROOT/pages/list-of-metrics.adoc:615]: I copied this section from IMap above. Same fixes should be done there

File: docs/modules/ROOT/pages/list-of-metrics.adoc [hunk 3]
```diff
@@ -433,6 +433,196 @@ This is because the cluster has to communicate with more members, which can add
...
+|`vectorCollection.totalSetLatency`
+|ms
+|Total latency of local set operations on the vector collection
+
+|`vectorCollection.totalSizeLatency`
+|ms
+|Total latency of local size operations on the vector collection
+
+3+a|
+The above `*latency` metrics are only measured for the members and they are not representing the overall performance of the cluster.
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/ROOT/pages/list-of-metrics.adoc:613]: `*latency` is a wildcard matching everything that ends with "latency", but without star it is also ok. but the same adjustments should be made for IMap metrics description

File: docs/modules/ROOT/pages/list-of-metrics.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/ROOT/pages/list-of-metrics.adoc:614]:
```suggestion
We recommend monitoring the average latency for each operation, for example, `vectorCollection.totalGetLatency / vectorCollection.getCount` and `vectorCollection.totalSetLatency / vectorCollection.setCount`.
```

[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/ROOT/pages/list-of-metrics.adoc]: changed to single expressions: https://github.com/hazelcast/hz-docs/pull/1171/commits/e2ec2a42f996b25786c6e2ff7eec1f717dbfdfaa

File: docs/modules/ROOT/pages/list-of-metrics.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/ROOT/pages/list-of-metrics.adoc]: Ah, I thought that was mistake with bold formatting perhaps - ignore my suggestion

File: docs/modules/ROOT/pages/list-of-metrics.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/ROOT/pages/list-of-metrics.adoc]: Thanks, that helps - "/" is not always used to mean divide but I guess in this context it's pretty clear (though I missed it on first skim reading).

**Review by oliverhowell — APPROVED** [review]

**Pull request merged**